api/middleware.go

```diff
@@ -18,12 +19,37 @@ const (
 	jwsSignatureHeaderName = "x-nf-sign"
 )
 
+type FunctionHooks map[string][]string
+
 type NetlifyMicroserviceClaims struct {
 	jwt.StandardClaims
-	SiteURL       string            `json:"site_url"`
-	InstanceID    string            `json:"id"`
-	NetlifyID     string            `json:"netlify_id"`
-	FunctionHooks map[string]string `json:"function_hooks"`
+	SiteURL       string        `json:"site_url"`
+	InstanceID    string        `json:"id"`
+	NetlifyID     string        `json:"netlify_id"`
+	FunctionHooks FunctionHooks `json:"function_hooks"`
+}
+
+func (f FunctionHooks) UnmarshalJSON(b []byte) error {
+	var raw map[string]interface{}
+	err := json.Unmarshal(b, &raw)
+	if err != nil {
+		return err
+	}
+
+	if f == nil {
+		f = make(map[string][]string)
+	}
+
+	for event, i := range raw {
+		switch val := i.(type) {
+		case []string:
+			f[event] = val
+		case string:
+			// Support legacy format for function hooks json.
+			f[event] = []string{val}
+		}
```

Comment:
wdyt about adding a default case so we don't silently ignore invalidly formatted hook payloads?
it could either return an error or log a warning

Reply:
New version will return an error from `json.Unmarshal` in this case :+1: